lib/json0.js

```diff
@@ -663,6 +663,61 @@ json.transformComponent = function(dest, c, otherC, type) {
   return dest;
 };
 
+json.transformPresence = function(presence, op, isOwnOp) {
+  if (!presence || !isArray(presence.p)) return null;
+  if (!op) return presence;
+
+  presence = clone(presence);
+  op = clone(op);
+
+  // Deletions should be treated as insertions: transforming
+  // by them should result in a no-op (null presence), so
+  // let's just pretend that deletions are oi to get the
+  // desired transform behaviour
+  for (var i = 0; i < op.length; i++) {
+    const component = op[i];
+    if ('od' in component) {
+      component.oi = component.od;
+      delete component.od;
+    }
+
+    if ('ld' in component) {
```

Comment:
I'm curious if this breaks the `json.transform(transformed, [component], 'right')` lower down, if you, say, have presence on index 4 of an array and do a `ld` on index 2.

Reply:
Great catch. Fixed with a path check.